### DISCUSSION TITLE
Integrazione Alexa skill e errore autenticazione

### DISCUSSION THREAD

**Discussion by Daximino** [indomus/forum#2586] [GENERALE - Discussioni generiche]
Buonasera,
Sto lottando da oggi pomeriggio con la skill di Alexa.
Premetto che da web quando accedo alla URL mio_dominio/api mi torna il messaggio "not found" se vado  invece all'indirizzo mio_dominio/api/alexa/smart_home mi dice "method not allowed".
In pratica l'errore che mi ritorna da applicazione Alexa quando faccio l'autenticazione mi dice:
Connessione senza successo
Il server per l'autorizzazione ha restituito il parametro error. Per maggiori dettagli, consulta https://tools.ietf.org/html/rfc6749#section-4.1.2.1.

Da che può dipendere?

**Comment by snakuzzo** [comment]
Quei messaggi sono corretti. È così che deve essere. Probabilmente il tuo problema è il token sulla lambda function (per caso hai cancellato qualche token da HA?)
Prova a rigenerarne uno nuovo e lo vai a sostituire nel config.json sulla lambda function su AWS, salvi e fai Deploy.
Se non sai/ricordi di cosa sto parlando, vatti a guardare la guida per haaska

**Reply by Daximino** [reply]
Generato un nuovo token, incollato su config.json, test effettuato con successo.
Quando vado da cellulare su app amazon alexa e lo connetto ad un utente mi restituisce lo stesso errore

**Reply by snakuzzo** [reply]
Verifica se hai errori sul log di HA quando fai le prove

**Reply by Daximino** [reply]
Non vedo errori sul log home assistant core riferiti ad Alexa.
Mi dicevano di controllare auth0 in quanto sono esposto con duckdns, come posso fare?

**Reply by snakuzzo** [reply]
Dicevano chi?
Auth0 cosa?

**Reply by Daximino** [reply]
Il problema potrebbe essere legato a OAuth?
L'applicazione di Amazon Alexa è mi riporta a questo sito: https://developer.amazon.com/en-US/docs/alexa/account-linking/add-account-linking.html

**Reply by snakuzzo** [reply]
Prova a seguire [questo consiglio](https://t.me/indomus_chat/475009) e facci sapere se così risolvi.

**Reply by Daximino** [reply]
Ho inserito quella riga ma mi ritorna lo stesso errore. Esiste un log che posso vedere? Il problema è legato ad Amazon non ad home assistant perché nei log di ha non arriva nulla

**Reply by Daximino** [reply]
Fatto altre prove ma non riesco a capire che cosa gli dà fastidio nell'autenticazione

**Reply by Daximino** [reply]
Ho risolto, Ho cambiato python mettendo la versione 3.6, ed adesso va
